### PR TITLE
Implement "Add Above" functionality in paragraphs

### PR DESCRIPTION
**Pull request by cienvaras**
# READY FOR REVIEW

## Summary
- Install [Paragraphs Editor Enhancements](https://www.drupal.org/project/paragraphs_ee) and [Paragraphs Features](https://www.drupal.org/project/paragraphs_features)
- Replace Paragraph Browser with the default Paragraphs widget
- Enable the "Add Above" option
- Refactor Paragraph Browser custom code to work with the new widget
- Alter the paragraphs widget to remove the paragraph icons and change the add paragraph buttons to match previous version

## Need Review By (Date)
09/11

## Urgency
medium

## Steps to Test
For both Traditional and Colorful:

1. Create new Flexible and Private pages. Confirm that all component fields use the default paragraphs widget with the enhanced modal provided by `paragraphs_ee` (see screenshot below)
2. Add a Collection component to the Flexible page. Confirm that it also uses the new widget/modal
3. Edit an existing page with multiple components. Click the contextual menu in one of the paragraphs and select "Add Above". Confirm that the paragraphs selector modal is displayed and you can add a new paragraph in the expected position

Screenshot 2024-09-04 at 10 32 10 PM

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by cienvaras** [comment]
@ahughes3 This one is ready for functional testing. However, there are a couple of pending things:

- There's still one Codeception error. It doesn't seem to be related with the changes on the paragraphs, but it's not present on other branches. I ran multiple tests and can confirm that all the errors related with the paragraph widget changes are fixed, so it would be great if we can get some help from SWS.
- This PR and #1574 will have merge conflicts because both changed the same code. Fixing the conflicts is relatively simple, but we'll need to merge the other PR before doing that.

It would be good if we can pass the PR to SWS for an initial review, even when there's pending work, to detect early any possible kickbacks and speed up the review process.

**Comment by joegl** [comment]
@cienvaras @ahughes3 Is the intention to replace `paragraphs_browser`? I've never seen `paragraph_ee` before and will have to spend time with it. It looks neat 👍

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by cienvaras** [comment]
> Is the intention to replace `paragraphs_browser`?

@joegl That's right, we're using the default paragraphs widget, with a better modal provided by `paragraphs_ee`. Paragraphs browser have some issues, one of them being not supporting the "Add Above" functionality for adding new paragraphs on top of existing ones.

Changing the widget required to update the Codeception tests, mostly updating selectors and texts. For some reason we're getting a weird error in one of the tests that doesn't seem to be related with the new widget:

```
There was 1 error:
1) FlexiblePageCest: Test spotlight slider
 Test  codeception/functional/Install/Content/FlexiblePageCest.php:testSpotlightSlider

  [TypeError] strpos(): Argument #1 ($haystack) must be of type string, null given
```

https://github.com/SU-HSDO/suhumsci/actions/runs/10795561917/job/29964351113?pr=1599

The error happens when adding an image to a spotlight component and doesn't seem to be a functional one, it's a PHP error. It's definitely related to the changes in the PR, as it's not present on other branches, but I haven't been able to pinpoint what's happening. Do you have any idea of what could be going on or any pointers on how to debug this?

**Comment by joegl** [comment]
> > Is the intention to replace `paragraphs_browser`?
> 
> @joegl That's right, we're using the default paragraphs widget, with a better modal provided by `paragraphs_ee`. Paragraphs browser have some issues, one of them being not supporting the "Add Above" functionality for adding new paragraphs on top of existing ones.
> 
> Changing the widget required to update the Codeception tests, mostly updating selectors and texts. For some reason we're getting a weird error in one of the tests that doesn't seem to be related with the new widget:
> 
> ```
> There was 1 error:
> 1) FlexiblePageCest: Test spotlight slider
>  Test  codeception/functional/Install/Content/FlexiblePageCest.php:testSpotlightSlider
> 
>   [TypeError] strpos(): Argument #1 ($haystack) must be of type string, null given
> ```
> 
> https://github.com/SU-HSDO/suhumsci/actions/runs/10795561917/job/29964351113?pr=1599
> 
> The error happens when adding an image to a spotlight component and doesn't seem to be a functional one, it's a PHP error. It's definitely related to the changes in the PR, as it's not present on other branches, but I haven't been able to pinpoint what's happening. Do you have any idea of what could be going on or any pointers on how to debug this?

I will take a look when I review. I noticed this PR is still in a draft state, is it meant to be?

**Comment by cienvaras** [comment]
> I noticed this PR is still in a draft state, is it meant to be?

@joegl No, it's ready, I just changed the state. I'm expecting some conflicts with #1574 in the update hook and config, but that should be easy to fix regardless of which PR gets merged first.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by cienvaras** [comment]
@joegl Albert mentioned that you had some concerns about the Paragraph Features module. We're adding it because it's a required dependency for Paragraph Enhanced Editor, but we're not using any specific feature of the module. We didn't found any possible problems with it, but let us know if you have any questions.

**Comment by joegl** [comment]
> @joegl Albert mentioned that you had some concerns about the Paragraph Features module. We're adding it because it's a required dependency for Paragraph Enhanced Editor, but we're not using any specific feature of the module. We didn't found any possible problems with it, but let us know if you have any questions.

Yea I noticed there were a couple open bug reports relating to the features added, but if they're not being used then it's not a problem 👍 

The other thing I noticed was the module has [a hard-coded requirement](https://www.drupal.org/project/paragraphs_features/issues/3353704) for the gin or claro themes for the add-in feature. I know the gin theme is a base theme for the current admin theme, and I wasn't sure if it would be compatible or not. But again, if this feature isn't even being used, it's a non-issue.

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install [hunk 1]
```diff
@@ -226,3 +226,66 @@ function hs_basic_page_update_10000() {
     }
   }
 }
+
```
Comment (joegl) [inline docroot/modules/humsci/hs_basic_page/hs_basic_page.install]: I would add a database update hook before this one to install both the modules. The below suggestion is for the `su_humsci_profile` but could be put here too (although I'd suggest keeping module install/uninstall hooks in the profile).

```suggestion
/**
 * Install paragraphs_ee and paragraphs_features modules.
 */
function su_humsci_profile_update_9709() {
  \Drupal::service('module_installer')->install(['paragraphs_features']);
  \Drupal::service('module_installer')->install(['paragraphs_ee']);
}
```

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install [hunk 2]
```diff
@@ -226,3 +226,66 @@ function hs_basic_page_update_10000() {
...
+  foreach ($fields as $field => $form_display_id) {
+    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display */
+    $form_display = $entity_form_display_storage->load($form_display_id);
+    if (empty($form_display)) {
+      continue;
+    }
+    $component = $form_display->getComponent($field);
+    if (empty($component)) {
+      continue;
+    }
```
Comment (joegl) [inline docroot/modules/humsci/hs_basic_page/hs_basic_page.install]:
```suggestion
  ];
  
  $entity_form_display_storage = \Drupal::entityTypeManager()->getStorage('entity_form_display');
  foreach ($fields as $field => $form_display_id) {
    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display */
    $form_display = $entity_form_display_storage->load($form_display_id);
    if (empty($form_display)) {
      continue;
    }
    
    $component = $form_display->getComponent($field);
    if (empty($component)) {
      continue;
    }
    
```

I'd add some spaces to break up code / improve legibility.

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install [hunk 3]
```diff
@@ -226,3 +226,66 @@ function hs_basic_page_update_10000() {
...
+    $component['type'] = 'paragraphs';
+    $title = $component['settings']['title'];
+    $title_plural = $component['settings']['title_plural'];
+    $default_paragraph_type = $component['settings']['default_paragraph_type'];
+    $component['settings'] = $base_settings;
+    $component['settings']['title'] = $title;
+    $component['settings']['title_plural'] = $title_plural;
+    $component['settings']['default_paragraph_type'] = $default_paragraph_type;
```
Comment (joegl) [inline docroot/modules/humsci/hs_basic_page/hs_basic_page.install]:
```suggestion
  $component_settings = array_intersect_key($component['settings'], array_flip(['title', 'title_plural', 'default_paragraph_type']));
    $settings = array_merge($component_settings, $base_settings);

    $component['type'] = 'paragraphs';
    $component['settings'] = $settings;
```

A couple array functions here could help simplify this.
* https://www.php.net/manual/en/function.array-intersect-key.php
* https://www.php.net/manual/en/function.array-flip.php
* https://www.php.net/manual/en/function.array-merge.php

File: docroot/modules/humsci/hs_admin/hs_admin.module [hunk 4]
```diff
@@ -191,14 +191,40 @@ function hs_admin_preprocess_page(&$variables) {
...
   for ($delta = 0; $delta <= $max_delta; $delta++) {
     if (isset($field_widget_complete_form['widget'][$delta]['top']['actions']['dropdown_actions']['duplicate_button'])) {
       $duplicate_button = &$field_widget_complete_form['widget'][$delta]['top']['actions']['dropdown_actions']['duplicate_button'];
       $duplicate_button['#ajax']['callback'] = 'su_humsci_profile_paragraphs_duplicate_callback';
+      // Remove icon from paragraph items in node edit form.
+      unset($field_widget_complete_form['widget'][$delta]['top']['type']['icon']);
     }
   }
+  // Change the add paragraph button in flexible page, private page and
+  // collection fields to match the previously used paragraph browser viersion.
```
Comment (joegl) [inline docroot/modules/humsci/hs_admin/hs_admin.module]:
```suggestion
  // Change the add paragraph button in flexible page, private page and
  // collection fields.
```
I wouldn't reference something that's being deprecated in a comment. Future coders may not understand the reference. I'd either drop it (like suggestion) or specify the actual behavior.

**Review by joegl — CHANGES_REQUESTED** [review]
This looks good to me, just a few suggestions and recommend installing the modules in the update hook.

I'll also investigate the test failure soon.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install (hunk 1 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_basic_page/hs_basic_page.install]: Done.

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install (hunk 2 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_basic_page/hs_basic_page.install]: Fixed.

File: docroot/modules/humsci/hs_basic_page/hs_basic_page.install (hunk 3 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_basic_page/hs_basic_page.install]: Good idea! Thanks for providing the links to the functions. Fixed now.

File: docroot/modules/humsci/hs_admin/hs_admin.module (hunk 4 above)
Comment (cienvaras) [reply docroot/modules/humsci/hs_admin/hs_admin.module]: Fixed.

**Comment by cienvaras** [comment]
@joegl Thanks for the feedback, all issues have been addressed.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by joegl** [comment]
Thanks @cienvaras ! I think we're close on this one besides tests. What I'd like to do is:

* Get #1574 completed and merged into the next release branch, and then merge the next release branch into this PR (#1574 contains a few tests fixes that should help this PR)
* Clean up any remaining test failures in this PR, if there are any.
* Merge this.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://english-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://english-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://english-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://english-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Review by joegl — APPROVED** [review]

**Comment by joegl** [comment]
@cienvaras I'm ready to merge this one but wanted you to get a chance to review the merge commits first 👍

**Comment by cienvaras** [comment]
@joegl Looks good, thanks! I just fixed an incorrect value in the update hook, but other than that everything works as expected.

**Comment by ahughes3** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1599-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-colorful-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://hs-traditional-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com
- https://english-ruwju0azgq716emqmttqub0wsc8pncct.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7

Visual Diffs:
- https://dashboard.tugboatqa.com/66bb9c706c7d83449ce87bb7/visualdiffs

**Pull request merged**